doc/source/dev/development_workflow.rst

```diff
@@ -150,6 +153,110 @@ been added to ``upstream`` that affect your work. In this case, follow the
 :ref:`rebasing-on-main` section of this document to apply those changes to
 your branch.
 
+.. _formatting-with-pre-commit:
+
+Automatic code formatting with pre-commit
+-----------------------------------------
+
+NumPy recommends that code follow PEP7/PEP8 guidelines, but does not currently
+enforce any formatters. However, the repository does come with a an optional
+pre-commit hook that performs automatic formatting and validation of changes,
+and using it is highly recommended. Tasks and checks that it runs include:
+
+* Format changed lines only of Python files to pep8 standard using `Darker
+  <darker_>`_, which wraps `black <black_>`_ and `isort <isort_>`_ and applies
+  them to the diff (this hook is enabled by default). Additionally, check only
+  these changed lines using `flake8 <flake8_>`_.
+
+* Format changed lines only of C and C++ source files to pep7 standard using a
+  Python wrapper for `ClangFormat versioning <clang_format_>`_ (this hook is
+  enabled by default).
+
+* Format and lint YAML shell files, and perform linting on Fortran, rst, Docker,
```

Comment:
What are YAML shell files? Do you mean shell commands in YAML files?

Reply:
Fixed in 794131fbbef4d3bf464a70ec400da985b9684dfd